audformat/core/utils.py

```diff
@@ -174,7 +175,7 @@ def concat(
         if not filewise.all():
             objs = [to_segmented_index(obj) for obj in objs]
     else:
-        _convert_single_level_multi_index(objs)
+        objs = _convert_single_level_multi_index(objs)
```

Comment:
Interesting, why did the code worked before without any errors?
Because we modified the objects in place?

Reply:
The type hint says we have a `Sequence`, which does not support item assignment, i.e. we have to convert the sequence to a list inside the function. And since we now have a new object we must return it.